keras_nlp/utils/text_generation.py

```diff
@@ -288,63 +298,91 @@ def token_probability_fn(inputs):
     ```
 
     """
-    if not tf.executing_eagerly():
-        raise RuntimeError(
-            "`keras_nlp.utils.beam_search` currently requires an eager "
-            "execution context. Please call `beam_search` outside "
-            "tf.function or run `tf.config.run_functions_eagerly(True)` to run "
-            "tf.function in eager mode."
-        )
     if num_beams <= 0:
         raise ValueError(
             f"`num_beams` should be strictly positive. Received: `num_beams={num_beams}`."
         )
-
     prompt = _validate_prompt(prompt)
 
-    if prompt.shape[-1] == 0:
-        raise ValueError(
-            "Length of `prompt` is 0, please provide a non-empty `prompt`."
-        )
-
     input_is_1d = prompt.shape.rank == 1
     if input_is_1d:
         prompt = prompt[tf.newaxis, :]
-    _validate_token_probability_fn(token_probability_fn, prompt)
 
-    batch_size, length = prompt.shape
+    batch_size, length = _get_prompt_shape(prompt)
+    prompt, mask = _pad_prompt(prompt, max_length)
+
+    vocab_size = _validate_token_probability_fn(
+        token_probability_fn, prompt, return_vocab_size=True
+    )
+
     if length >= max_length:
         return tf.squeeze(prompt) if input_is_1d else prompt
+    # Initialize beam with shape `(batch_size, num_beams, length)`.
+    beams = tf.repeat(tf.expand_dims(prompt, axis=1), num_beams, axis=1)
+
+    # Initialize beams_prob with shape `(batch_size, num_beams)`
+    beams_prob = tf.zeros([batch_size, 1], dtype=tf.float32)
+    if num_beams > 1:
```

Comment:
what does num_beams == 1 even mean? can we just route to greedy or something simpler in that case at the top of this fn?

Reply:
Makes sense. Will do